msticpy/context/azure/sentinel_core.py

```diff
@@ -54,97 +111,102 @@ class MicrosoftSentinel(
-        ws_name: Optional[str] = None,
+        subscription_id: Optional[str] = None,
+        resource_group: Optional[str] = None,
+        workspace_name: Optional[str] = None,
         **kwargs,
     ):
         """
         Initialize connector for Azure APIs.
 
         Parameters
         ----------
-        res_id : str, optional
+        resource_id : str, optional
             Set the Sentinel workspace resource ID you want to use, if not specified
-            defaults will be looked for or details can be passed separately with functions.
+            defaults will be looked for or details can be passed separately with functions,
+            by default None
+            `res_id` is an alias for resource_id.
         connect : bool, optional
             Set true if you want to connect to API on initialization, by default False
         cloud : str, optional
             Specify cloud to use, overriding any configuration value.
             Default is to use configuration setting or public cloud if no
             configuration setting is available.
-        sub_id : str, optional
+        subscription_id : str, optional
             If not specifying a resource ID the Subscription ID of the Sentinel Workspace
             by default None
-        res_grp : str, optional
+            `sub_id` is an alias for subscription_id
+        resource_group : str, optional
             If not specifying a resource ID the Resource Group name of the
             Sentinel Workspace, by default None
-        ws_name : str, optional
+            `res_grp` is an alias for resource_group
+        workspace_name : str, optional
             If not specifying a resource ID, the Workspace name of the
             Sentinel Workspace, by default None
-        workspace : str, optional
-            Alias of ws_name
+            `ws_name` and `workspace` are aliases for workspace_name
+
+        Notes
+        -----
+        There are multiple ways to set the default workspace for the Microsoft Sentinel
+        class:
+        1. Specify a full Azure resource ID for the workspace in the `resource_id` parameter.
+        2. Specify the subscription ID and resource group and workspace name in the
+        `subscription_id`, `resource_group` and `workspace` parameters.
+        3. Specify only a workspace name in the `workspace` parameter. This will read
+        the workspace details from the msticpyconfig configuration file.
 
         """
         super().__init__(connect=False, cloud=cloud)
+
+        init_kwargs = _map_legacy_param_names(**kwargs)
+        if resource_id:
+            init_kwargs[_RES_ID] = resource_id
+        if subscription_id:
+            init_kwargs[_SUB_ID] = subscription_id
+        if resource_group:
+            init_kwargs[_RES_GRP] = resource_group
+        if workspace_name:
+            init_kwargs[_WS_NAME] = workspace_name
+
+        self._default_settings = self._set_ws_defaults(_get_ws_defaults, **init_kwargs)
```

Comment:
This is probably worth a type annotation:

`self._default_settings: Callable[..., SentinelInstanceDetails] = self._set_ws_defaults(_get_ws_defaults, **init_kwargs)`